ddpui/api/dashboard_chat_api.py

```diff
@@ -0,0 +1,2493 @@
+            ai_messages.append(AIMessage(role=msg.role, content=msg.content, metadata=msg.metadata))
+
+        # Generate session ID for this request
+        session_id = str(uuid.uuid4())
+        request_time = timezone.now()
+
+        # Get the latest user message for logging
+        user_message = None
+        for msg in reversed(payload.messages):
+            if msg.role == "user":
+                user_message = msg.content
+                break
+
+        # ENHANCED: Use SmartChatProcessor for intent-aware handling.
+        # Data-query execution is only enabled when org has AI data sharing enabled,
+        # but dashboard explanations can still be generated from metadata even without data sharing.
+        enhanced_response = None
+        org_settings = OrgSettings.objects.filter(org=orguser_obj.org).first()
+        if user_message and org_settings:
+            try:
+                from ddpui.core.ai.smart_chat_processor import SmartChatProcessor
+
+                # Initialize smart chat processor
+                smart_processor = SmartChatProcessor()
+
+                # Process the message with enhanced capabilities
+                enhanced_response = smart_processor.process_enhanced_chat_message(
+                    message=user_message,
+                    org=orguser_obj.org,
+                    dashboard_id=dashboard_id,
+                    dashboard_context=context,
+                    user_context={},
+                    enable_data_query=bool(org_settings.ai_data_sharing_enabled),
+                )
+
+                # If query was executed successfully, return the enhanced response
+                if enhanced_response.query_executed and enhanced_response.data_results:
+                    logger.info(
+                        f"Dashboard {dashboard_id} - Smart query executed successfully for: {user_message[:100]}"
+                    )
+
+                    # Log the interaction
+                    log_ai_chat_conversation(
+                        org=orguser_obj.org,
+                        user=orguser_obj.user,
+                        user_prompt=user_message,
+                        ai_response=enhanced_response.content,
+                        request_timestamp=request_time,
+                        response_timestamp=timezone.now(),
+                        dashboard_id=dashboard_id,
+                        chart_id=payload.selected_chart_id,
+                        session_id=session_id,
+                    )
+
+                    # Log metering
+                    log_ai_chat_metering(
+                        org=orguser_obj.org,
+                        user=orguser_obj.user,
+                        model_used="smart_query_processor",
+                        prompt_tokens=len(user_message.split()) * 1.3,  # Rough estimate
+                        completion_tokens=len(enhanced_response.content.split())
+                        * 1.3,  # Rough estimate
```

Comment:
how is the number tokens computed , for both prompt & the response ?
where is `1.3` coming from